docs/Test.md

```diff
@@ -0,0 +1,150 @@
+# Test
+
+Tests serve many different purposes. They can:
+
+- Validate the requirements/expectations of stakeholders are met
+- That an end user can achieve their goal
+- Act as documenatation/reference material
+- Confirm integration works as expected between areas of function
+- Validate we do not regress functionality or behaviour we care about
+
+We want to enable and achieve all of these goals in the UI testing done in this
+repo. To this end, we require, depending on the functionality being added or
+changed, different types of tests. The types of test expected, and the
+rationale for these are detailed below.
+
+## Style of test
+
+We have two styles of test - `Behavioural` and `Functional`.
```

Comment:
Instead Behavioral style and Functional style do we just want to go with Behavior Drive Development and Test Driven Development?

Reply:
Fair point, I'll update these to be more familiar